packages/core/src/common/command.ts

```diff
@@ -73,12 +74,12 @@ export interface CommandHandler {
      * Execute this handler.
      */
     // tslint:disable-next-line:no-any
-    execute(...args: any[]): any;
+    execute(...args: any[]): MaybePromise<any>;
     /**
      * Test whether this handler is enabled (active).
      */
     // tslint:disable-next-line:no-any
-    isEnabled?(...args: any[]): boolean;
+    isEnabled?(...args: any[]): MaybePromise<boolean>;
```

Comment:
All clients (even transitive, e.g. via `getActiveHandler`)  should be analyzed and should be changed to handle async `isEnabled`.

For instance, `KeybindingRegistry.isActive` will always return `true` for promises.